addons/userrights/userrightsprofile.class.inc.php

```diff
@@ -738,8 +738,10 @@ protected function GetUserActionGrant($oUser, $sClass, $iActionCode)
 		// load and cache permissions for the current user on the given class
 		//
 		$iUser = $oUser->GetKey();
-		$aTest = @$this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
-		if (is_array($aTest)) return $aTest;
+		if (isset($this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode])){
+			$aTest = $this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
+			if (is_array($aTest)) return $aTest;
+		}
```

Comment:
Not sure if the array test is still needed?

```suggestion
		if (isset($this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode])){
			return $this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
		}
```

Otherwise, you could just add this in one go:
```suggestion
		if (isset($this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode]['permission'])){
			return $this->m_aObjectActionGrants[$iUser][$sClass][$iActionCode];
		}
```

Reply:
after a quick searck, $aTest must be an array at this point of the code. iis_array seems useless yes. 

but i prefer that this code remains defensive (until types are reenforced by method signatures?)... one day someone may add another entry which is not an array anymore (by mistake). at least it will not break anything...

Reply:
Then my second suggestion could be used ;)

Reply:
this is not a code that i know very well. 2nd proposal seems ok to me. fonctionnaly i cannot approve or reject on my own.

@Molkobain ?

Reply:
TBH I never went to this part of the code, so I would rather just keep the defensive test as proposed by Olivier. No real need to go further (yet).

Reply:
Closing conversation as everyone agrees